### ISSUE TITLE
Wrong shape used for `Function` prototype in `fig4` of `this &... / chapter 6`?

### ISSUE THREAD

**Issue by FlorianKempenich** [getify/You-Dont-Know-JS#1492]
#### `fig4` of `this &... / chapter 6` - [Section in question](https://github.com/getify/You-Dont-Know-JS/blob/master/this%20%26%20object%20prototypes/ch6.md#mental-models-compared)

Hi,

First of all thanks for the amazing content! 

I found what I think is an error, but considering I first learned about prototypes something like 48h ago I wouldn't be surprised if it's actually correct and I simply mis-understood something 🙂 

Anyways, in [`fig4.png`](https://github.com/getify/You-Dont-Know-JS/blob/master/this%20%26%20object%20prototypes/fig4.png), shouldn't the `[[ Prototype ]]` of the `Function()` "constructor" be depicted as an object?  
##### Here's a picture which explains what I mean
![fig4withquestion](https://user-images.githubusercontent.com/10831649/61997457-88049f80-b099-11e9-984b-35336b1728c2.png)

Thanks!

Best,
Florian Kempenich

**Comment by getify** [comment]
It is indeed actually a function, from which all other functions descend.

**Comment by FlorianKempenich** [comment]
Hi,

Thanks for your answer!

After reading your comment I had a hard time wrapping my head around what that means, and why that would be interesting to do so. So basically 2 questions remained:
1. How is it even possible?
2. Why is it interesting to do so?

I quickly remembered that `Functions` are `Objects` as well, so at least it started to make sense and question **1** was answered, but there was still question **2**. Eventually I ended up on the official specification for the [Function.prototype](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-function.prototype) and I saw this:
![image](https://user-images.githubusercontent.com/10831649/62006912-0e26f180-b13f-11e9-9e3b-255176a719da.png)

So that answers question **2** as well 🙂

Thanks for your content! Seriously! I would never have pushed so far without the support of your research, fantastic work!
